k8s-install/canal.yaml

```diff
@@ -0,0 +1,218 @@
+# This manifest installs the per-node agents, as well
+# as the CNI plugins and network config on 
+# each master and worker node in a Kubernetes cluster.
+kind: DaemonSet
+apiVersion: extensions/v1beta1
+metadata:
+  name: canal-node
+  namespace: kube-system
+  labels:
+    kubernetes.io/cluster-service: "true"
+    k8s-app: canal-node
+spec:
+  selector:
+    matchLabels:
+      kubernetes.io/cluster-service: "true"
+      k8s-app: canal-node
+  template:
+    metadata:
+      labels:
+        kubernetes.io/cluster-service: "true"
+        k8s-app: canal-node
+    spec:
+      hostNetwork: true
+      containers:
+        # Runs the flannel daemon to enable vxlan networking between
+        # container hosts.
+        - name: flannel-daemon
+          image: quay.io/coreos/flannel:0.5.5
+          env:
+            # The location of the etcd cluster.
+            - name: FLANNELD_ETCD_ENDPOINTS
+              valueFrom:
+                configMapKeyRef:
+                  name: canal-config
+                  key: etcd_endpoints
+            # The interface flannel should run on. 
+            - name: FLANNELD_IFACE
+              valueFrom:
+                configMapKeyRef:
+                  name: canal-config
+                  key: flannel_iface
+            # Perform masquerade on traffic leaving the pod cidr.
+            - name: FLANNELD_IP_MASQ
```

Comment:
I've not quite got my head around whether this is always the right thing to do, have you?